torch/csrc/distributed/c10d/init.cpp

```diff
@@ -3161,9 +3161,52 @@ such as `dist.all_reduce(tensor, async_op=True)`.
         Arguments:
           tensors(List[torch.Tensor]): List of tensors we want to hash.
       )");
-  module.def("_dump_nccl_trace", []() {
-    return py::bytes(::c10d::dump_nccl_trace());
-  });
+  module.def(
+      "_get_collective_trace",
```

Comment:
I made this API with the hopes that there might be cases where we want do debug collectives but not NCCL comm. So this is just a convenient function.
My thought is that collective debugging v/s NCCL comm debugging evolve independently so this is an attempt to start separating these out.

In my mind, the collected data looks as follows:
{
COMMON STUFF (pg_config)
COLLECTIVE_STUFF (entries)
NCCL_COMM_STUFF(a map)
}